analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -185,7 +184,16 @@ private IEnumerable<ExplodedNode> ProcessOperation(ExplodedNode node)
                     yield return node.CreateNext(node.Operation.Parent is null && node.Block.BranchValue != node.Operation.Instance ? postProcessed.State.ResetOperations() : postProcessed.State);
                 }
             }
-            if (ExceptionCandidate(node.Operation) is { } exception && node.Block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
+
+            foreach (var explodedNode in ExceptionSuccessors(node, ExceptionCandidate(node.Operation.Instance)))
+            {
+                yield return explodedNode;
+            }
+        }
+
+        private IEnumerable<ExplodedNode> ExceptionSuccessors(ExplodedNode node, ExceptionState exception)
+        {
+            if (exception is { } && node.Block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
```

Comment:
_Nitpick:_ A tiny bit more suitable, because we don't need to declare the `exception` anymore
```suggestion
            if (exception is not null && node.Block.EnclosingRegion(ControlFlowRegionKind.Try) is { } tryRegion)
```